utils/utils.go

```diff
@@ -81,88 +32,44 @@ func Untar(ctx context.Context, source, target string) error {
 	return nil
 }
 
-func ScrapeErrorMessage(message string, regExpression string) string {
-
-	re, err := regexp.Compile(regExpression)
-	scrapped := re.FindStringSubmatch(message)
-
-	if scrapped == nil || err != nil || len(scrapped) < 2 {
-		// couldn't scrape, return generic error
-		message = "Error executing operation"
-	} else {
-		// return scrapped response
-		message = scrapped[1]
-	}
-
-	return message
-}
-
-func CheckReturnCode(res int, mesg []byte) {
-	if res >= 300 {
-
-		message := string(mesg[:])
-		log.Debugf("IMCO API response: %s", message)
-
-		f := func(c rune) bool {
-			return c == ',' || c == ':' || c == '{' || c == '}' || c == '"' || c == ']' || c == '['
-		}
-
-		// check if response is a web page.
-		if strings.Contains(message, "<html>") {
-			scrapResponse := "<title>(.*?)</title>"
-			message = ScrapeErrorMessage(message, scrapResponse)
-		} else if strings.Contains(message, "{\"errors\":{") {
-			scrapResponse := "{\"errors\":(.*?)}"
-
-			message = ScrapeErrorMessage(message, scrapResponse)
-			result := strings.Split(message, ",")
-			if result != nil && len(result) >= 1 {
-				message = result[0]
-			}
-			// Separate into fields with func.
-			fields := strings.FieldsFunc(message, f)
-			message = strings.Join(fields[:], " ")
-
-		} else if strings.Contains(message, "{\"error\":") {
-			scrapResponse := "{\"error\":\"(.*?)\"}"
-			message = ScrapeErrorMessage(message, scrapResponse)
-		}
-		// if it's not a web page or json-formatted message, return the raw message
-		log.Fatal(fmt.Sprintf("HTTP request failed: (%d) [%s]", res, message))
-	}
-}
-
 // CheckStandardStatus return error if status is not OK
-func CheckStandardStatus(status int, mesg []byte) error {
+func CheckStandardStatus(status int, response []byte) error {
 
 	if status < 300 {
 		return nil
 	}
 
-	message := string(mesg[:])
-
-	f := func(c rune) bool {
-		return c == ',' || c == ':' || c == '{' || c == '}' || c == '"' || c == ']' || c == '['
-	}
-
-	if strings.Contains(message, "{\"errors\":{") {
-		scrapResponse := "{\"errors\":(.*?)}"
-
-		message = ScrapeErrorMessage(message, scrapResponse)
-		result := strings.Split(message, ",")
-		if result != nil && len(result) >= 1 {
-			message = result[0]
+	// default, raw, not parsing
+	message := string(response[:])
+
+	var responseContent map[string]interface{}
+	err := json.Unmarshal(response, &responseContent)
+	if err == nil {
+		if responseContent["errors"] != nil {
+			message = ""
+			for key, value := range responseContent["errors"].(map[string]interface{}) {
+				subMessages := value.([]interface{})
+				composedMsg := ""
+				if len(subMessages) > 1 {
+					// several error messages per field
+					for _, m := range subMessages {
+						if composedMsg != "" {
+							composedMsg = fmt.Sprintf("%s,%s", composedMsg, m)
+						} else {
+							composedMsg = fmt.Sprintf("%s", m)
+						}
```

Comment:
Actually, I'd keep strings.Join here, the whole body of the line 50 loop can reduced to:
```Go
  composedMsg := strings.Join(subMessages, ",")
  message = fmt.Sprintf("%s#%s:%s", message, key, composedMsg)
```

Reply:
good point :100: but since 'Join' requires an []string array, a conversion is needed. At any case, I'd prefer you proposal, and based on it, this proposal: 


```
subMessages := make([]string, len(value.([]interface{})))
for i, v := range value.([]interface{}) {
      subMessages[i] = fmt.Sprint(v)
}
composedMsg := strings.Join(subMessages, ",")
message = fmt.Sprintf("%s#%s:%s", message, key, composedMsg)
```

Reply:
Maybe a strings.Builder could be used

Reply:
Updated code based on proposal